feat: Add Chat Trigger node

alexgrozav opened the pull request:
<img width="1496" alt="image" src="https://github.com/n8n-io/n8n/assets/6179477/6885df39-cc3a-48cf-8cbc-3d522e2428aa">


[AI_Chat_Trigger.json](https://github.com/n8n-io/n8n/files/12870898/AI_Chat_Trigger.json)

janober committed:
:zap: Make supplemental labels easier to read

janober committed:
:zap: Apply same style for pulling connections of non-main connectors

jesper-bylund committed:
Adding anthropic SVG instead of PNG

jesper-bylund committed:
Set all the langchain icons to currentColor

janober committed:
:zap: Hardcode colors in icons for now until we can make currentColor work

janober committed:
:zap: Fix offset positions of newly created nodes

OlegIvaniv committed:
feat(editor): Implement input mapping mode for config nodes (#7259)

Github issue / Community forum post (link here to close automatically):

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

janober committed:
:zap: Rename InMemory Vector Store node

janober committed:
:zap: Fix langchain package name

janober committed:
:zap: Add doc links to nodes

janober committed:
Revert ":zap: Fix langchain package name"

This reverts commit 1775dc49e47c6a692007a59a6aadc31b2a45b22e.

janober committed:
:zap: Fix anthropic icon

alexgrozav committed:
feat(editor): Add various AI Tool UX improvements (no-changelog) (#7264)

<img width="1496" alt="image"
src="https://github.com/n8n-io/n8n/assets/6179477/9be76be3-9bb9-4b32-ae83-824ea03579fe">
<img width="797" alt="image"
src="https://github.com/n8n-io/n8n/assets/6179477/9bf505a7-2c61-40b0-bc37-9a949c23145e">
<img width="1496" alt="image"
src="https://github.com/n8n-io/n8n/assets/6179477/8473b669-081d-423f-bc88-ce7a7ff3a888">

---------

Co-authored-by: Jan Oberhauser <jan.oberhauser@gmail.com>

janober committed:
:zap: Some node name changes

OlegIvaniv committed:
feat(core): Google PaLM/Cohere/TensorFlow/HuggingFace embeddings, GooglePaLM LM/Chat (#7265)

Github issue / Community forum post (link here to close automatically):

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>
Co-authored-by: Jan Oberhauser <jan.oberhauser@gmail.com>

janober committed:
Merge branch 'master' into ai-tool-creation-breaking

janober committed:
:zap: Update langchain to version 0.154.0

OlegIvaniv committed:
:bug: Use `helpers.getBinaryDataBuffer` to correctly get binary data with `filesystem` N8N_DEFAULT_BINARY_DATA_MODE

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

janober committed:
:zap: Rename langchain nodes package (breaking-change)

janober committed:
Merge branch 'ai-tool-creation-breaking' of github.com:n8n-io/n8n into ai-tool-creation-breaking

OlegIvaniv committed:
feat(editor): Restructure AI node creator view (#7276)

Github issue / Community forum post (link here to close automatically):

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

janober committed:
:zap: Improve descriptions

janober committed:
:zap: Change Splitter icons

OlegIvaniv committed:
:art: Improve filtering of models for OpenAI Chat/LM

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

OlegIvaniv committed:
:bug: Fix sorting of agents/chain in AI root view

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

StarfallProjects committed:
Copy roundup (#7278)

OlegIvaniv committed:
:sparkles: Add miscellaneous category

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

janober committed:
:zap: Add additional description to vector store nodes

janober committed:
:zap: Also change the Text Splitter icon in the menu

StarfallProjects committed:
Update packages/@n8n/nodes-langchain/nodes/agents/OpenAiFunctionsAgent/OpenAiFunctionsAgent.node.ts

OlegIvaniv committed:
Use gpt-3.5-turbo-instruct as default OpenAI LM model

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

janober committed:
:zap: Remove Vector QA chain and just have a single QA one instead

janober committed:
:zap: Fix display of log for Vector Store Retriever

OlegIvaniv committed:
:bug: Fix keydown events propagation when chat is open

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

janober committed:
feat: Combine all Agents into one

janober committed:
Merge branch 'ai-tool-creation-breaking' of github.com:n8n-io/n8n into ai-tool-creation-breaking

OlegIvaniv committed:
feat(core): AI Nodes text changes (#7302)

Github issue / Community forum post (link here to close automatically):

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

OlegIvaniv committed:
:bug: Fix OpenAIFunction and ReAct agents

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

OlegIvaniv committed:
feat(editor):  Implement beta release UI changes  (#7291)

Github issue / Community forum post (link here to close automatically):

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

janober committed:
:zap: Remove StructuredOutput Chain

OlegIvaniv committed:
:goal_net: Check for correct models when using Conversational & OpenAi Functions agents

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

janober committed:
:zap: Update package information

janober committed:
:zap: Set version number

janober committed:
Merge branch 'master' into ai-tool-creation-breaking

janober committed:
:zap: Remove run mode and added default prompt to ChainRetrievalQA

OlegIvaniv committed:
:bug: Fix SQL Agent credentials

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

OlegIvaniv committed:
feat(core): Allow to override agent/chain system messages (#7306)

Github issue / Community forum post (link here to close automatically):

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

janober committed:
:zap: Fix option order

janober committed:
:zap: Fix data transfer + improve error reporting in ToolWorkflow

OlegIvaniv committed:
:bug: Refresh AI Logs in Chat on new message

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

OlegIvaniv committed:
💄 Prevent LM Chat message overflow & add logs title

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

OlegIvaniv committed:
:bug: Fix InputPanel noOutputData state for config node nodes

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

janober committed:
:zap: Add AI nodes also to trigger view

OlegIvaniv committed:
:speech_balloon: Update summarization chain types description

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

OlegIvaniv committed:
:lipstick: Fix node-options context menu positions & increase hover area

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

alexgrozav committed:
feat: Add  AI tool creation chatbot widget (#7137)

Co-authored-by: Jan Oberhauser <janober@users.noreply.github.com>

janober committed:
:zap: Fix issue that ToolWorkflow did not display errors correctly

Joffcom committed:
feat(core): Support some of the Postgres SSL opens in AI Agent (#7309)

This PR adds support for some of the SSL options when using Postgres in
the SSL agent, Tested against a standard docker postgres image with no
ssl support and Neon.tech which requires SSL.

janober committed:
Merge branch 'ai-tool-creation-breaking' of github.com:n8n-io/n8n into ai-tool-creation-breaking

janober committed:
:zap: Fix issue that ToolWorkflow did not display an helpful error on
invalid workflow ID

janober committed:
feat: Add MultiQuery Retriever

janober committed:
feat: Add Workflow Retriever

janober committed:
:zap: Change namespace in ItemListOutputParser

janober committed:
:zap: Fix some lint issues

janober committed:
:zap: Update n8n-nodes-langchain package version to 0.0.2

janober committed:
:zap: Remove package.json from eslint to fix error for now

OlegIvaniv committed:
:bug: Fix preview environment canvas rendering issues

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

netroy committed:
Merge remote-tracking branch 'origin/master' into ai-tool-creation-breaking

netroy committed:
enable lazy-loading for n8n-nodes-langchain

janober committed:
:zap: Hide "Execute Node" button also for dynamic inputs

janober committed:
:zap: Fix issue with code nodes and main input

StarfallProjects committed:
Debs editing (#7317)

Round-up of copy edits.
Includes making the Retrieval Q&A node names consistent (they were
previously different in the nodes panel and the node) This will be
confusing for users, especially ones working with templates, but also
anyone using the docs.

janober committed:
:zap: Another fix for hide "Execute Node" button

janober committed:
:zap: Prepare for correct parameter resolution in sub-nodes

janober committed:
:zap: Add license file to @n8n/chat

janober committed:
:zap: Fix @n8n/chat exmple workflow to work with new agent nodes

janober committed:
:zap: Very minor code improvement

janober committed:
:zap: Fix issue that sometimes AI-logs did not get saved

janober committed:
:zap: Display helpful error message if input/prompt/text is empty

janober committed:
:zap: Change default tool names to be OpenAI Functions compatible

OlegIvaniv committed:
fix(core): Use regular LLM chain if there's no output parser (#7330)

Github issue / Community forum post (link here to close automatically):

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

OlegIvaniv committed:
:bug: Fix RunDataAi not updating based on run & error message when closing NDV

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

janober committed:
:zap: Update n8n-nodes-langchain version + slighly update README

StarfallProjects committed:
docs: Add LangChain section to README (#7332)

@janober the links need changing when we know what the final branch will
be

janober committed:
Merge branch 'ai-tool-creation-breaking' of github.com:n8n-io/n8n into ai-tool-creation-breaking

janober committed:
Merge branch 'master' into ai-beta

janober committed:
Merge branch 'master' into ai-beta

janober committed:
:zap: Update README links for nodes + chat

netroy committed:
Merge remote-tracking branch 'origin/master' into ai-beta

OlegIvaniv committed:
fix(editor): Fix rare dragging exception and unify LLM chain outputKey (#7341)

Github issue / Community forum post (link here to close automatically):

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

janober committed:
bug(editor): Fix filter not used when inputs defined as expression

alexgrozav committed:
feat: Update chat widget to work without bundler (no-changelog) (#7352)

Github issue / Community forum post (link here to close automatically):

janober committed:
Merge branch 'master' into ai-beta

janober committed:
:zap: Remove no longer used connection type AiVectorRetriever

alexgrozav committed:
feat: add chat trigger node

OlegIvaniv committed:
refactor(core): Refactor InMemory Vector Store to support Inser & Load  (#7362)

Github issue / Community forum post (link here to close automatically):

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

janober committed:
:zap: Make sure that it always auto selects a node with data

janober committed:
:zap: Remove old debug console.log

janober committed:
:zap: Make $parameter expressions of sub-nodes resolve correctly in UI

janober committed:
:zap: Add baseURL support to OpenAI nodes

janober committed:
:zap: Fix zep authentication

janober committed:
Merge branch 'master' into ai-beta

OlegIvaniv committed:
:bug: Expose this.logger to Langchain code sandbox & fix example connection key

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

OlegIvaniv committed:
fix(core): Fetch Ollama models dynamically (#7395)

Github issue / Community forum post (link here to close automatically):
https://community.n8n.io/t/ollama-node-add-api-pull-ai-beta/31380

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

OlegIvaniv committed:
:hammer: Fix langchain packaged dev watcher and add dev:ai to root package.json

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

alexgrozav committed:
feat: add chat trigger

alexgrozav committed:
feat: add chat trigger auth

alexgrozav committed:
Merge branch 'ai-beta' into chat-trigger-node

janober committed:
:zap: Clean up interface of getInputConnectionData

netroy committed:
Merge remote-tracking branch 'origin/master' into ai-beta

janober committed:
:zap: Update n8n-nodes-langchain package version to 0.2.0

netroy committed:
cleanup @n8n/nodes-langchain dependencies

netroy committed:
Merge remote-tracking branch 'origin/master' into ai-beta

alexgrozav committed:
fix: Fix readme workflow link in @n8n/chat package (#7438)

Github issue / Community forum post (link here to close automatically):

janober committed:
Merge branch 'master' into ai-beta

OlegIvaniv committed:
fix: Always stringify workflow tool outpus (#7447)

Github issue / Community forum post (link here to close automatically):

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

netroy committed:
Merge remote-tracking branch 'origin/master' into ai-beta

janober committed:
:zap: Add back dependencies

netroy committed:
Merge remote-tracking branch 'origin/master' into ai-beta

netroy committed:
Merge remote-tracking branch 'origin/master' into ai-beta

OlegIvaniv committed:
feat(core): Improve structured schema & add maxIteration for agents (#7407)

Github issue / Community forum post (link here to close automatically):

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

OlegIvaniv committed:
feat(core): Add support to include system/ai/human pre-messages for LLM Chain (#7475)

Github issue / Community forum post (link here to close automatically):

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>
Co-authored-by: Jan Oberhauser <jan.oberhauser@gmail.com>

OlegIvaniv committed:
feat(core): Add support for metadata setting and filtering from vector stores (#7491)

Github issue / Community forum post (link here to close automatically):
- https://community.n8n.io/t/supabase-insert-outputs/31831/5
-

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

janober committed:
Merge branch 'master' into ai-beta

janober committed:
Merge branch 'master' into ai-beta

janober committed:
Merge branch 'master' into ai-beta

janober committed:
:zap: Add missing dependency to make csv loading work

OlegIvaniv committed:
fix(core): Fix basic LLM chain failing if query contains curly brackets (#7553)

Github issue / Community forum post (link here to close automatically):

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

StarfallProjects committed:
update vector store links (#7562)

Companion to https://github.com/n8n-io/n8n-docs/pull/1711

OlegIvaniv committed:
feat(core): Add support for Huggin Face inference endpoints to LM and embs (#7563)

Github issue / Community forum post (link here to close automatically):
-
https://discord.com/channels/832547762716278804/1151891018149543936/1168666212042022933

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

janober committed:
Merge branch 'ai-beta' into chat-trigger-node

janober committed:
:zap: Fix option order and default values

janober committed:
:zap: Rename from 'Additional Fields' to 'Options'

janober committed:
:zap: Remove not needed await

OlegIvaniv committed:
feat(core): Merge & refactor vector store nodes & allow for similiarity retrieval (#7539)

Github issue / Community forum post (link here to close automatically):

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

janober committed:
Merge branch 'master' into ai-beta

OlegIvaniv committed:
feat(core): Add support for AWS Bedrock LMs & Embeddings (#7581)

Github issue / Community forum post (link here to close automatically):

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

OlegIvaniv committed:
refactor: Rearrange Supabase & Pinecone vector store fields (#7589)

Github issue / Community forum post (link here to close automatically):

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>
Co-authored-by: Jan Oberhauser <jan.oberhauser@gmail.com>

janober committed:
Merge branch 'master' into ai-beta

janober committed:
:zap: Fix output labels in dark mode

janober committed:
Merge branch 'master' into ai-beta

OlegIvaniv committed:
docs: Update vector-store docs links (#7594)

Github issue / Community forum post (link here to close automatically):

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

janober committed:
:zap: Update n8n-nodes-langchain package version to 0.3.0

janober committed:
Merge branch 'ai-beta' of github.com:n8n-io/n8n into ai-beta

janober committed:
Merge branch 'master' into ai-beta

StarfallProjects committed:
docs: Update two docs links (#7638)

Part of this effort to fix some 404s:
https://n8nio.slack.com/archives/C03KQFPMVEV/p1699358947262899

OlegIvaniv committed:
Merge branch 'master' into ai-beta

michael-radency committed:
feat(core): Langchain nodes tweaks (#7640)

Github issue / Community forum post (link here to close automatically):

OlegIvaniv committed:
feat(core): AI usability improvements (#7627)

## Show a notice with example workflow
![CleanShot 2023-11-08 at 11 35
20](https://github.com/n8n-io/n8n/assets/12657221/60881cdd-ebe0-47b5-b7f3-bfb3b909930e)

## Show connection hint in NDV if sub-node is not connected

https://github.com/n8n-io/n8n/assets/12657221/27417869-d87b-4932-9983-6cf7b0edd0cc

## Merge JSON and Binary document loaders and add support for mime-type
detection
The JSON and Binary document loaders are now merged into "Default Data
Loader". This allows to select type of data(JSON or Binary). For binary,
it defaults to auto-detecting of mime-type. JSON mode additionally
supports populating documents via an expression.


https://github.com/n8n-io/n8n/assets/12657221/058c5850-b83e-4e23-9957-eee8a1bbb2aa

## Improve sub-nodes errors rendering and usability


https://github.com/n8n-io/n8n/assets/12657221/f15c4b8a-efcc-49f1-89e5-cfb49131e1a7

## Some text updates

Github issue / Community forum post (link here to close automatically):

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>
Co-authored-by: Jan Oberhauser <jan.oberhauser@gmail.com>

gandreini committed:
Fixed chat messages colors for dark mode.

MasonGeloso committed:
feat: Add Plan and Execute Agent Type (#7632)

**Plan and Execute Agent Type Implementation**
Implementation of LangChain's experimental Plan and Execute agent type:
https://js.langchain.com/docs/modules/agents/agent_types/plan_and_execute


https://github.com/n8n-io/n8n/assets/6699213/826f66c8-a314-4a89-8940-291317e39479

---------

Co-authored-by: Mason Geloso <hone@Masons-Mac-mini.local>

alexgrozav committed:
feat: Add AI features unit and e2e tests (#7513)

- Adds unit tests for @n8n/chat package
<img width="726" alt="image"
src="https://github.com/n8n-io/n8n/assets/6179477/b9372f26-c476-4721-acd9-9586b333d25e">

- Adds unit tests for various editor-ui files
- Adds e2e tests for AI Nodes and new input types
- Refactor editor-ui for unit testing
- Pinia triggers a strange error when stores are imported from an index
file, this PR fixes it
  ```
FAIL src/components/__tests__/WorkflowSettings.spec.ts [
src/components/__tests__/WorkflowSettings.spec.ts ]
  TypeError: Cannot read properties of undefined (reading '$id')
❯
../../node_modules/.pnpm/pinia@2.1.6_typescript@5.2.2_vue@3.3.4/node_modules/pinia/dist/pinia.mjs:1803:26
      1801|     return stores.reduce((reduced, useStore) => {
      1802|         // @ts-expect-error: $id is added by defineStore
1803| reduced[useStore.$id + mapStoreSuffix] = function () {
         |                          ^
      1804|             return useStore(this.$pinia);
      1805|         };
   ❯ Module.mapStores 

../../node_modules/.pnpm/pinia@2.1.6_typescript@5.2.2_vue@3.3.4/node_modules/pinia/dist/pinia.mjs:1801:19
   ❯ src/components/WorkflowSettings.vue:454:6
   ❯ src/components/__tests__/WorkflowSettings.spec.ts:1:31
  ```

mutdmour committed:
chore: revert icon changes

OlegIvaniv committed:
Merge branch 'master' into ai-beta

janober committed:
fix: Prevent hanging AI executions (#7680)

Github issue / Community forum post (link here to close automatically):
- https://github.com/n8n-io/n8n/issues/7666#issuecomment-1805586041
- https://community.n8n.io/t/structured-output-parser-issue/31984/6

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>
Co-authored-by: Oleg Ivaniv <me@olegivaniv.com>
Co-authored-by: कारतोफ्फेलस्क्रिप्ट™ <aditya@netroy.in>

netroy committed:
remove custom CORS support from webhook node

OlegIvaniv committed:
:wrench: Increase the default timeout for OpenAI models to 1min

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

OlegIvaniv committed:
feat: Enable to input image in Basic LLM Chain & render it in the execution log (#7739)

![CleanShot 2023-11-16 at 23 42
05](https://github.com/n8n-io/n8n/assets/12657221/6db3c6c4-af38-4c00-b6cb-e93fdac365a8)

Github issue / Community forum post (link here to close automatically):
-
https://community.n8n.io/t/please-add-support-of-the-new-openai-features/32525
(Partially)

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

janober committed:
Merge branch 'master' into ai-beta

janober committed:
Merge branch 'master' into ai-beta

OlegIvaniv committed:
:wrench: Add Anthropic Claudia 2.1 model option

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

OlegIvaniv committed:
:bug: Correctly escape curly brackets when using chat prompts

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

OlegIvaniv committed:
Merge branch 'master' into ai-beta

OlegIvaniv committed:
Merge branch 'master' into ai-beta

alexgrozav committed:
chore: merge ai-beta

OlegIvaniv committed:
feat(editor): Floating NDV connected nodes (#7776)

https://github.com/n8n-io/n8n/assets/12657221/76f36721-5f99-49fa-93f2-c42c443e9dfa

Github issue / Community forum post (link here to close automatically):

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

OlegIvaniv committed:
:bug: Fix GPT vision image detail when using URL

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

alexgrozav committed:
fix: fix auth order and webhook urls

alexgrozav committed:
feat: add notice about n8n chat trigger in chat embed modal

janober committed:
:zap: By default do not support loading messages of previous sessions

janober committed:
:zap: Make things clearer and only provide required options to page

OlegIvaniv committed:
Merge branch 'master' into ai-beta

OlegIvaniv committed:
🔀 Fix merging issues

OlegIvaniv committed:
:bug: Fix sub-nodes error rendering

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

netroy committed:
Merge remote-tracking branch 'origin/master' into ai-beta

netroy committed:
Revert "fix: Prevent hanging AI executions (#7680)"

This reverts commit 11aa3b2c11ebdbd731c824a46acdcf2542961760.

netroy committed:
update build targets

netroy committed:
Merge remote-tracking branch 'origin/master' into ai-beta

netroy committed:
lintfix

netroy committed:
Merge remote-tracking branch 'origin/master' into ai-beta

OlegIvaniv committed:
✅ Fix editor unit tests store imports

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

netroy committed:
Merge remote-tracking branch 'origin/master' into ai-beta

OlegIvaniv committed:
:wrench: Enable MD support for textLoader

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

netroy committed:
switch back to the success notification on manual execution cancellation

netroy committed:
Merge remote-tracking branch 'origin/master' into ai-beta

netroy committed:
Merge remote-tracking branch 'origin/master' into ai-beta

netroy committed:
Merge remote-tracking branch 'origin/master' into ai-beta

netroy committed:
fix editor tests

michael-radency committed:
fix(core): Tweaks for Some AI nodes (no-changelog) (#7774)

- Tweak the format of the info boxes in nodes that are shown at the top
of sub-nodes when they’re not connected
- In nodes that are chat compatible, rename the default expression in
the prompt param from `$json.input` to `$json.chat_input`
- When executing a manual chat trigger and there’s no chain executed,
improve the error text
- Better Vector Store error messages
- Improve window buffer memory description
- Use RLC every time we’re loading a list from an API for:
  - OpenAI model (’model’ param)
  - OpenAI embedding (’model’ param)
  - Pinecone vector store
- Improve window buffer memory description
- Add an info box to auto-fixing output parser
- Change wording of the output pane of sub-node when there’s no
execution, and remove the link there. Plus remove the edit button

---------

Co-authored-by: Oleg Ivaniv <me@olegivaniv.com>

netroy committed:
Merge remote-tracking branch 'origin/master' into ai-beta

OlegIvaniv committed:
feat: OpenAI assistant (#7795)

Add `OpenAI Assistant` node with support for both OpenAI tools and LC
tools
Github issue / Community forum post (link here to close automatically):
-
https://community.n8n.io/t/please-add-support-of-the-new-openai-features/32525/16

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>
Co-authored-by: Jan Oberhauser <jan.oberhauser@gmail.com>

janober committed:
Merge branch 'ai-beta' into chat-trigger-node

janober committed:
Merge branch 'master' into chat-trigger-node

janober committed:
Merge branch 'master' into chat-trigger-node

janober committed:
:zap: Simplify required Chat Trigger setup

alexgrozav committed:
chore: merge master

alexgrozav committed:
fix: update Chat Trigger node size when loading session

alexgrozav committed:
feat: update example @n8n/chat workflow

alexgrozav committed:
fix: use node type instead of node name for size

janober committed:
Merge branch 'master' into chat-trigger-node

alexgrozav committed:
feat: update chat widget configuration

alexgrozav committed:
Merge branch 'chat-trigger-node' of github.com:n8n-io/n8n into chat-trigger-node

alexgrozav committed:
feat: add chat images

alexgrozav committed:
feat: add n8n user metadata to hosted chat

alexgrozav committed:
chore: bump package version

alexgrozav committed:
feat: update chat README.md

alexgrozav committed:
chore: bump package version

alexgrozav committed:
fix: add fallback for showWelcomeScreen param

alexgrozav committed:
fix: fix metadata

alexgrozav committed:
fix: add metadata to session request and fix template

alexgrozav committed:
fix: update template chat version to latest

alexgrozav committed:
fix: fix getting shwoWelcomeScreen option

alexgrozav committed:
Merge branch 'master' into chat-trigger-node

janober committed:
:zap: Fix some lint issues

janober committed:
:zap: Small fix + change

alexgrozav committed:
chore: merge master

alexgrozav committed:
test: fix broken unit tests

janober committed:
Merge branch 'master' into chat-trigger-node

janober committed:
:zap: Add button to open chat

janober committed:
:zap: Rename to make clearer distinction to the external chat

janober committed:
:zap: Make $json work in Sub-Nodes connected to Trigger-Nodes

janober committed:
:zap: Temporary change chat session ID to be identical with widget

janober committed:
:zap: Change some descriptions

janober committed:
:zap: Hide panel content if mode testChat is selected

janober committed:
:zap: Do not display Chat-Button in Webhook-Mode

janober committed:
:zap: Create automatically Chat-Trigger instead of Manual Chat Trigger if missing

janober committed:
Merge branch 'master' into chat-trigger-node

janober committed:
:zap: Fix on test

StarfallProjects commented on packages/@n8n/nodes-langchain/nodes/trigger/ChatTrigger/ChatTrigger.node.ts:
```suggestion
		description: 'Runs the workflow when an n8n generated webchat is submitted',
```

StarfallProjects commented on packages/@n8n/nodes-langchain/nodes/trigger/ChatTrigger/ChatTrigger.node.ts:
```suggestion
		activationMessage: 'You can now make calls to your production chat URL.',
```

StarfallProjects commented on packages/@n8n/nodes-langchain/nodes/trigger/ChatTrigger/ChatTrigger.node.ts:
Although: should this actually be webhook URL?

StarfallProjects commented on packages/@n8n/nodes-langchain/nodes/trigger/ChatTrigger/ChatTrigger.node.ts:
```suggestion
					"The Chat Trigger has two modes: test and production. <br /> <br /> <b>Use test mode while you build your workflow</b>. Click the 'Test Step' button, then fill out the test chat that opens in a popup tab. The executions will show up in the editor.<br /> <br /> <b>Use production mode to run your workflow automatically</b>. <a data-key=\"activate\">Activate</a> the workflow, then make requests to the production URL. Then every time there's a chat submission via the Production Chat URL, the workflow will execute. These executions will show up in the executions list, but not in the editor.",
```

StarfallProjects commented on packages/@n8n/nodes-langchain/nodes/trigger/ChatTrigger/ChatTrigger.node.ts:
```suggestion
					"The Chat Trigger has two modes: test and production. <br /> <br /> <b>Use test mode while you build your workflow</b>. Click the 'Test Step' button, then fill out the test chat that opens in a popup tab. The executions will show up in the editor.<br /> <br /> <b>Use production mode to run your workflow automatically</b>. <a data-key=\"activate\">Activate</a> the workflow, then make requests to the production URL. Then every time there's a chat submission via the Production Chat URL, the workflow will execute. These executions will show up in the executions list, but not in the editor.",
```

StarfallProjects commented on packages/@n8n/nodes-langchain/nodes/trigger/ChatTrigger/ChatTrigger.node.ts:
```suggestion
				displayName: 'Open built-in chat',
```

StarfallProjects commented on packages/@n8n/nodes-langchain/nodes/trigger/ChatTrigger/ChatTrigger.node.ts:
The workflow doesn't execute until the user hits 'Send', right?

OlegIvaniv committed:
feat: Add new versions of AI nodes with new chat trigger defaults (#8110)

- Add a new version of `text` and `prompt` parameters for entities that
assume use of chat trigger to corresponding default(`={{ $json.chatInput
}}`, )
- Add a new version of `sessionId` for memory nodes to use the new
`$json.sessionId` property
- Use `chatInput` as input key when starting wf with debug chat, also
added `sessionId` while keeping `chat_history` for compatibility sake

---------

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

alexgrozav committed:
feat: Update Chat Trigger node (no-changelog) (#8099)

## Summary
> Describe what the PR does and how to test. Photos and videos are
recommended.

Address Chat trigger node review
https://www.notion.so/n8n/Chat-trigger-review-ef6bdb355eec43aeb1dc5592f33c1d87

## Related tickets and issues
> Include links to **Linear ticket** or Github issue or Community forum
post. Important in order to close *automatically* and provide context to
reviewers.

https://linear.app/n8n/issue/N8N-7029/chat-trigger-node

## Review / Merge checklist
- [x] PR title and summary are descriptive. **Remember, the title
automatically goes into the changelog. Use `(no-changelog)` otherwise.**
([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [x] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up
ticket created.
- [x] Tests included.
> A bug is not considered fixed, unless a test is added to prevent it
from happening again.
   > A feature is not complete without tests.

alexgrozav committed:
Update packages/@n8n/nodes-langchain/nodes/trigger/ChatTrigger/ChatTrigger.node.ts

Co-authored-by: Deborah <deborah@starfallprojects.co.uk>

alexgrozav committed:
Update packages/@n8n/nodes-langchain/nodes/trigger/ChatTrigger/ChatTrigger.node.ts

Co-authored-by: Deborah <deborah@starfallprojects.co.uk>

janober committed:
Merge branch 'master' into chat-trigger-node

alexgrozav committed:
fix: add unbuild to dependencies

alexgrozav committed:
chore: merge master

alexgrozav committed:
chore: merge master

alexgrozav committed:
chore: merge master

OlegIvaniv committed:
🔀Resolve merge conflicts in `workflow.service` and `TestWebhooks`

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

alexgrozav committed:
fix: adjust expected node search count in e2e test

alexgrozav committed:
fix: fix langchain e2e tests

OlegIvaniv commented on packages/@n8n/nodes-langchain/nodes/trigger/ChatTrigger/templates.ts:
This allows XSS since we can just enter any arbitrary HTML into the authentication parameter:
![CleanShot 2024-01-04 at 17 03 27](https://github.com/n8n-io/n8n/assets/12657221/d7fdcdc5-4e61-417f-9099-91963302cecf)
![CleanShot 2024-01-04 at 17 03 42](https://github.com/n8n-io/n8n/assets/12657221/0a0dec79-3ad3-4789-9987-98ff49e5aa9a)

OlegIvaniv commented on packages/@n8n/nodes-langchain/utils/EpubLoader.ts:
I think this might have been accidental merging error, we deleted this loader some time ago

OlegIvaniv commented on packages/editor-ui/src/components/ChatEmbedModal.vue:
🫠

OlegIvaniv reviewed (changes requested):
Left a few comments for discussion but awesome job! 🔥 Can't wait to get this merged

alexgrozav committed:
fix: fix xss vulnerability in authentication param

alexgrozav committed:
chore: merge master

alexgrozav commented on packages/@n8n/chat/tsconfig.json:
Done.

alexgrozav commented on packages/@n8n/nodes-langchain/nodes/trigger/ChatTrigger/ChatTrigger.node.ts:
I don't think so. Removed.

alexgrozav commented on packages/@n8n/nodes-langchain/nodes/trigger/ChatTrigger/ChatTrigger.node.ts:
Nope. Removed.

alexgrozav commented on packages/@n8n/nodes-langchain/nodes/trigger/ChatTrigger/ChatTrigger.node.ts:
Makes sense. Thanks for the suggestion.

alexgrozav commented on packages/@n8n/nodes-langchain/nodes/trigger/ChatTrigger/templates.ts:
😱 Great catch! Sanitized.

alexgrozav commented on packages/@n8n/nodes-langchain/utils/EpubLoader.ts:
Very likely. Deleted.

alexgrozav commented on packages/editor-ui/src/components/ChatEmbedModal.vue:
😭

alexgrozav committed:
fix: update pnpm-lock.yaml

krynble reviewed (dismissed):
I could only check the BE code - the `NodeExecuteFunctions` file as well as a couple changes to the workflow package. It all seems ok.

Did a manual test and nothing seems to be broken, so from what I can tell, this seems ok.

alexgrozav committed:
Merge branch 'master' into chat-trigger-node

alexgrozav committed:
chore: merge master

alexgrozav committed:
fix: sanitize showWelcomeScreen param

alexgrozav committed:
fix: update chat build command to avoid running twice in ci

alexgrozav committed:
fix: improve user input sanitization

OlegIvaniv reviewed (dismissed):
Thanks for addressing ally my points. LGTM!

alexgrozav committed:
Merge branch 'master' into chat-trigger-node

OlegIvaniv committed:
Replace `CHAT_TRIGGER_NODE_NAME` with `MANUA_CHAT_TRIGGER_NODE_NAME` in 30-langchain e2e spec

Signed-off-by: Oleg Ivaniv <me@olegivaniv.com>

OlegIvaniv reviewed (approved):
Pushed a small fix for the failing Langchain E2Es, let's hope it'll run green now 🤞
